src/realm/obj.cpp

```diff
@@ -1936,6 +1936,84 @@ bool Obj::remove_backlink(ColKey col_key, ObjLink old_link, CascadeState& state)
+                auto linking_obj_dictionary = linking_obj.get_dictionary_ptr(c);
+                auto pos = linking_obj_dictionary->find_any(ObjLink{other.get_table()->get_key(), other.get_key()});
+                REALM_ASSERT(pos != realm::npos);
+                Mixed key = linking_obj_dictionary->get_key(pos);
+                linking_obj_dictionary->insert(key, ObjLink{m_table->get_key(), get_key()});
+            }
+            else if (c.get_type() == col_type_Mixed && linking_obj.get_any(c).get_type() == type_TypedLink) {
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set_any(c, ObjLink{m_table->get_key(), get_key()});
+            }
+            else if (c.get_type() == col_type_Link) {
+                // Single link
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set(c, get_key());
+            }
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::handle_multiple_backlinks_during_schema_migration()
+{
+    REALM_ASSERT(!m_table->get_primary_key_column());
+    auto copy_links = [this](ColKey col) {
+        auto opposite_table = m_table->get_opposite_table(col);
+        auto opposite_column = m_table->get_opposite_column(col);
+        auto backlinks = get_all_backlinks(col);
+        for (auto backlink : backlinks) {
+            auto obj = m_table->create_object();
+            obj.copy(*this);
+            auto linking_obj = opposite_table->get_object(backlink);
+            fix_linking_object_during_schema_migration(linking_obj, obj, opposite_column);
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::copy(const Obj& other)
+{
+    // To be used carefully since it could potentially duplicate user data.
+    // since Obj::copy is for now basically only used for handling schema migration when there are multiple
+    // incoming links to some embedded table, this method is basically harmless, but if Obj::assign get's exposed,
+    // we will have to have 2 versions of it, one designed for migrations (in which links to embedded objects are
+    // duplicated) and one for handling normal copies (where an expecption is thrown if an outgoing link to an
+    // embedded table is found).
+
+    // handle correctly the case in which the value to copy is an outgoing link to an embedded table.
+    auto handle_object_to_copy = [this](Mixed value) -> Mixed {
+        TableRef target_table;
+        ObjKey target_obj_key;
+        if (is_outgoing_embedded_link(value, target_table, target_obj_key)) {
+            auto copy_target_object = target_table->create_linked_object();
+            auto target_obj = target_table->get_object(target_obj_key);
+            copy_target_object.copy(target_obj);
```

Comment:
Calling `copy` recursively works, but there is a risk of stack overflow when the depth of recursion is proportional to user data. Can you change this to use an iterative solution?

Reply:
OK I can change the algorithm, however I wonder how really deep can this chain of linked embedded objects could actually be in practise. I feel like we might never hit a stack overflow. However, I agree that we should play safe.

Reply:
Right now the limit is 100 levels of nesting, but this could change in the future. https://www.mongodb.com/docs/manual/reference/limits/

Reply:
This has now been resolved.